### PR TITLE
Add documentation links configuration files

### PR DESCRIPTION
**Pull request by DonOmalVindula**
### Purpose
This PR adds configurations to allow documentation links to be served from a single configuration file for better maintainability.

### Related Issues
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- None

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/app.tsx [hunk 1]
```diff
@@ -27,12 +27,13 @@ import {
...
 import { I18n, I18nModuleOptionsInterface } from "@wso2is/i18n";
 import {
-    Code,
     ChunkErrorModal,
+    Code,
     NetworkErrorModal,
     SessionManagementProvider,
     SessionTimeoutModalTypes
 } from "@wso2is/react-components";
+import { DocumentationProvider } from "@wso2is/react-components/src";
```
Comment (brionmario) [inline apps/console/src/app.tsx:36]:
```suggestion
import { DocumentationProvider } from "@wso2is/react-components";
```

File: apps/console/src/app.tsx [hunk 2]
```diff
@@ -43,6 +44,7 @@ import { initializeAuthentication } from "./features/authentication";
 import { EventPublisher, PreLoader } from "./features/core";
 import { ProtectedRoute } from "./features/core/components";
 import { Config, getBaseRoutes } from "./features/core/configs";
+import documentationLinks from "./features/core/configs/documentation";
```
Comment (brionmario) [inline apps/console/src/app.tsx:47]: Lets make `documentationLinks` a named export.
```suggestion
import documentationLinks from "./features/core/configs";
```

File: apps/console/src/extensions/configs/documentation.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ConsoleDocumentationLinksInterface } from "@wso2is/react-components/src/components/documentation/models";
```
Comment (brionmario) [inline apps/console/src/extensions/configs/documentation.ts:19]:
```suggestion
import { ConsoleDocumentationLinksInterface } from "@wso2is/react-components";
```

Comment (brionmario) [inline apps/console/src/extensions/configs/documentation.ts:19]: Let's keep this on the app side.
```suggestion
import { ConsoleDocumentationLinksInterface } from "@wso2is/react-components/src/components/documentation/models";
```

File: apps/console/src/extensions/configs/models/documentation.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export type DocumentationLinksExtensionInterface = Record<string, unknown>
```
Comment (brionmario) [inline apps/console/src/extensions/configs/models/documentation.ts:1]: Missing license header.
```suggestion
export type DocumentationLinksExtensionInterface = Record<string, unknown>;
```

File: apps/console/src/features/core/models/documentation.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+// DO NOT SHORTEN THE IMPORT PATH as it could lead to circular dependencies.
```
Comment (brionmario) [inline apps/console/src/features/core/models/documentation.ts:1]: Missing license header.

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/core/models/documentation.ts (hunk 5 above)
Comment (DonOmalVindula) [reply apps/console/src/features/core/models/documentation.ts:19]: Fixed with 642a464ebd9d74c5bde502d428ed8c8d82e0e211

File: apps/console/src/extensions/configs/models/documentation.ts (hunk 4 above)
Comment (DonOmalVindula) [reply apps/console/src/extensions/configs/models/documentation.ts]: Fixed with 642a464ebd9d74c5bde502d428ed8c8d82e0e211

File: apps/console/src/extensions/configs/documentation.ts (hunk 3 above)
Comment (DonOmalVindula) [reply apps/console/src/extensions/configs/documentation.ts]: Fixed with 8be4530f6408250e5393750b9dbd6a422bc4367f

File: apps/console/src/app.tsx (hunk 2 above)
Comment (DonOmalVindula) [reply apps/console/src/app.tsx]: Fixed with 8be4530f6408250e5393750b9dbd6a422bc4367f

File: apps/console/src/app.tsx (hunk 1 above)
Comment (DonOmalVindula) [reply apps/console/src/app.tsx]: Fixed with 8be4530f6408250e5393750b9dbd6a422bc4367f

**Pull request merged**